subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -616,13 +617,16 @@ int nrf_cloud_rest_location_get(struct nrf_cloud_rest_context *const rest_ctx,
 	req.body = payload;
 
 	/* Make REST call */
-	ret = do_rest_client_request(rest_ctx, &req, &resp, true, true);
+	ret = do_rest_client_request(rest_ctx, &req, &resp, true, !request->disable_response);
 
 	if (ret) {
 		goto clean_up;
 	}
 
-	if (result) {
+	if (result && request->disable_response) {
+		LOG_WRN("A result struct is provided but location response is disabled");
```

Comment:
Can you word the warning to use active voice?  e.g.:
`The cloud responded with a result struct yet the device did not request it.`

Reply:
that is not what is happening here.
the user set the `disable_response` flag AND provided `result` instead of passing NULL.

Reply:
hmm.  i suppose i could just remove the new `disable_response` flag and use the presence of `result` as the flag...